app/hotels/src/allHotels/searchForm/SearchForm.js

```diff
@@ -50,16 +52,57 @@ class SearchForm extends React.Component<Props> {
   handleCheckoutChange = (date: string) =>
     this.handleDateChange(date, 'checkout');
 
-  handleDateChange = (datestring: string, type: 'checkin' | 'checkout') => {
-    const date = moment(datestring, DISPLAY_DATE_FORMAT).toDate();
+  handleDateChange = (datestring: string, type: DateInputType) => {
+    const date = moment(datestring, DISPLAY_DATE_FORMAT);
+    const checkin =
+      type === 'checkin' ? date : moment(this.props.search.checkin);
+    const checkout =
+      type === 'checkout' ? date : moment(this.props.search.checkout);
+    const whatToAdjust = type === 'checkin' ? 'checkout' : 'checkin';
+    const dates = this.getValidDates(checkin, checkout, whatToAdjust);
```

Comment:
This part of the code is also not super readable. Especially because of ternary conditions like `'checkin' ? 'checkout' : 'checkin'`... 😊

Reply:
fixed